gluten-core/src/main/scala/org/apache/gluten/extension/columnar/OffloadSingleNode.scala

```diff
@@ -232,6 +233,35 @@ case class OffloadFilter() extends OffloadSingleNode with LogLevelUtil {
   }
 }
 
+case class OffloadWindow() extends OffloadSingleNode with LogLevelUtil {
+  import OffloadOthers._
+  private val replace = new ReplaceSingleNode()
+
+  override def offload(plan: SparkPlan): SparkPlan = plan match {
+    case window: WindowExec =>
+      if (TransformHints.isNotTransformable(window)) {
+        return window
+      }
+
+      val transformer = replace.doReplace(window)
+      val newChild = transformer.children.head match {
```

Comment:
Oops, let's either add an assertion like `assert(transformer.children.size == 1)`, or cast it to a unary node.

Reply:
ok，got it